File: Ent_UsAdmin/AgregarGastosLegales.vb
Imports System.ComponentModel
Imports System.Data.SqlClient

Public Class AgregarGastosLegales
    Public idCredito As String

    Private Sub AgregarGastosLegales_Load(sender As Object, e As EventArgs) Handles MyBase.Load

    End Sub

    Private Sub BackgroundGastos_DoWork(sender As Object, e As System.ComponentModel.DoWorkEventArgs) Handles BackgroundGastos.DoWork
        iniciarconexionempresa()
        Dim comandoGastos As SqlCommand
        Dim consultaGastos As String
        consultaGastos = "insert into GastosLegales values('" & Now.ToString("yyyy-MM-dd") & "','" & idCredito & "','" & txtMonto.Text & "','" & txtConcepto.Text & "','" & nmusr & "','" & nm_completeusr & "')"
        comandoGastos = New SqlCommand
        comandoGastos.Connection = conexionempresa
        comandoGastos.CommandText = consultaGastos
        comandoGastos.ExecuteNonQuery()

    End Sub

    Private Sub btn_actualizar_Click(sender As Object, e As EventArgs) Handles btn_actualizar.Click
        Cargando.Show()
        Cargando.MonoFlat_Label1.Text = "Insertando Gastos"
        Cargando.TopMost = True
        BackgroundGastos.RunWorkerAsync()
    End Sub

    Private Sub BackgroundGastos_RunWorkerCompleted(sender As Object, e As RunWorkerCompletedEventArgs) Handles BackgroundGastos.RunWorkerCompleted
        Me.Invoke(Sub()
                      InformacionLegal.BackgroundClientes.RunWorkerAsync()

                  End Sub)
        Cargando.Close()
        Me.Close()

    End Sub

    Private Sub Panel1_Paint(sender As Object, e As PaintEventArgs) Handles Panel1.Paint

    End Sub

    Private Sub Panel1_MouseDown(sender As Object, e As MouseEventArgs) Handles Panel1.MouseDown
        If e.Button = MouseButtons.Left Then
            MoveForm(Me)
        End If
    End Sub
End Class

File: Ent_UsAdmin/Legal/AgregarGastosLegales.cs
using System;
using System.ComponentModel;
using System.Data.SqlClient;
using System.Windows.Forms;

namespace ConfiaAdmin
{

    public partial class AgregarGastosLegales
    {
        public string idCredito;

        public AgregarGastosLegales()
        {
            InitializeComponent();
        }

        private void AgregarGastosLegales_Load(object sender, EventArgs e)
        {

        }

        private void BackgroundGastos_DoWork(object sender, DoWorkEventArgs e)
        {
            Module1.iniciarconexionempresa();
            SqlCommand comandoGastos;
            string consultaGastos;
            consultaGastos = "insert into GastosLegales values('" + DateTime.Now.ToString("yyyy-MM-dd") + "','" + idCredito + "','" + txtMonto.Text + "','" + txtConcepto.Text + "','" + Module1.nmusr + "','" + Module1.nm_completeusr + "')";
            comandoGastos = new SqlCommand();
            comandoGastos.Connection = Module1.conexionempresa;
            comandoGastos.CommandText = consultaGastos;
            comandoGastos.ExecuteNonQuery();

        }

        private void btn_actualizar_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.Cargando.Show();
            My.MyProject.Forms.Cargando.MonoFlat_Label1.Text = "Insertando Gastos";
            My.MyProject.Forms.Cargando.TopMost = true;
            BackgroundGastos.RunWorkerAsync();
        }

        private void BackgroundGastos_RunWorkerCompleted(object sender, RunWorkerCompletedEventArgs e)
        {

            Invoke(new Action(() => My.MyProject.Forms.InformacionLegal.BackgroundClientes.RunWorkerAsync()));
            My.MyProject.Forms.Cargando.Close();
            Close();

        }

        private void Panel1_Paint(object sender, PaintEventArgs e)
        {

        }

        private void Panel1_MouseDown(object sender, MouseEventArgs e)
        {
            if (e.Button == MouseButtons.Left)
            {
                Module_resize.MoveForm(this);
            }
        }
    }
}